File: python/decorator/StarbuzzCoffee/decorator/condiments/Whip.py
from component.Beverage import Beverage
from decorator.CondimentDecorator import CondimentDecorator


class Whip(CondimentDecorator):
    __beverage = Beverage()

    def __init__(self, beverage):
        super().__init__()
        self.__beverage = beverage

    def get_description(self):
        return self.__beverage.get_description() + ", Whip"

    def cost(self):
        return self.__beverage.cost() + 0.22

File: ruby/decorator/StarbuzzCoffee/decorator/condiments/whip.rb
require './decorator/condiment_decorator'

class Whip < CondimentDecorator

  def initialize(beverage)
    @@beverage = beverage
  end

  def get_description
    super
    @@beverage.get_description + ", Whip"
  end

  def cost
    super
    @@beverage.cost + 0.10
  end

  private
  @@beverage = Beverage.new

end